hammer_time/hammer_time.py

```diff
@@ -33,12 +34,37 @@ def perform(client):
         remove_and_wait(client, new_status.iter_new_machines(old_status))
 
 
+def choose_machine(client):
+    """Choose a machine from the client's model at random.
+
+    :param client: The ModelClient to get machines for.
+    :return: a machine-id.
+    :raises: InvalidActionError if there are no machines to choose from.
+    """
+    status = client.get_status()
+    machines = list(m for m, d in status.iter_machines(containers=False))
+    if len(machines) == 0:
+        raise InvalidActionError('No machines to choose from.')
+    return choice(machines)
+
+
+class RebootMachineAction:
+    """Action that reboots a machine."""
+
+    def generate_parameters(client):
+        return {'machine_id': choose_machine(client)}
+
+    def perform(client, machine_id):
+        """Add and remove many containers using the cli."""
+        old_status = client.get_status()
+        client.juju('ssh', (machine_id, 'sudo', 'reboot'), check=False)
```

Comment:
I think the correct approach here might be 

1) confirm that the command executed successfully (I think that the websocket api will return an error if the command doesn't complete successfully; using the cli, you can simply check for a return code > 0).

2) Watch to make sure that the agent stays healthy. Matrix did this implicitly with its health check (the health check is actually the place where matrix typically raises a test failure -- chaos/glitch doesn't raise errors unless some action raises an actual Exception).

Here, I guess you have the option of adding a Thread or Process that does the same thing as matrix's health, or adding a loop here that tries to determine whether the reboot happened before executing.